Brookes Instagram Data Report. Hypothesis: My top liked account will be a fashion account. 

The theoretical applications of my hypothesis are that I have taken up a liking towards fashion and "influencers" fashion posts. My discover page consists of mostly fashion/clothing related posts so I believe that this is my top liked account. The statistical applications of my hypothesis are that I believe that the algorithm knows my top data and knows that I will interact with it on the discover page since. 

The data included in the liked_posts.json file is appropriate for testing your hypothesis because it shows all of the data I have collected since the beginning of my entire instagram account, which was early 2010s. Since the algorithm has been collecting my data for such a long time, the algorithm has seen all of my interests throughout the years.

Instagram data analysts collected this data.
This data is collected in order for business accounts to analyze their insights.
This data may be reliable because there is most likely a group of data analysts and people in marketing who work together for Instagram that must know all this infromation so that they can promote users interests, as well as ads to the user. 
This data may be unreliable because it is most likely overall in the hands of a computer to gather the overall information. Tehcnology isn't always perfect and has a great possibility of error, there could be missing data. 

In [3]:
import pandas as pd
import json 

My first step is importing my current instagram data downloaded onto my computer in a jsnon file using the "with open" function, wich opens the file following the current pathway on my computer and opening it as j, which is just the name I picked. Then I am able to save the file by using the .load function and storing it into my_data. 

In [4]:
with open ("/Users/brookemoncol/Desktop/brooke.moncol_20230905/likes/liked_posts.json") as j: 
    my_data = json.load(j)

My next step is to create a data frame. I do this by using the function pd.DataFrame, then accessing the file with my data saved, which is stored in "my_data" and accessing the likes_media_likes folder. I am then sotring this into a new function called likes_data.  

In [5]:
likes_data = pd.DataFrame(my_data['likes_media_likes'])

The third step is to create a new column called timestamps, which shows my likes per account. I do this by taking my_data and accessing the the likes_media_likes folder. Taking the 0 element in the list for string_list_data and timestamp . I then use list comprehension and store that newly created information in a new function called new_list. Then I am able to take the new column timestamp and add it to the new_list function. 

In [6]:
likes = my_data['likes_media_likes'][0]['string_list_data'][0]['timestamp']

In [7]:
new_list = [x['string_list_data'][0]['timestamp'] for x in my_data['likes_media_likes']]

In [9]:
likes_data['timestamp'] = new_list

My final step is to sort the likes per account from the beginning, including the top 10, as I feel a fashion account will be majority of the accounts shown. I do this by using the groupby function, which groups the data by the title, and use the count function to show the string_list_data. I store this information in a new function called grouped. I then use the sort_values function with my previous grouped function to sort the number of likes to go from descending, to see which is my topped liked account, and to use the .head function as well to show my topped 10 likes. 

In [65]:
grouped = likes_data.groupby('title').count()

In [84]:
grouped.sort_values('timestamp', ascending = False).head(10)

string_list_data  timestamp
title                                             
cavs                               1145       1145
cohmedy                            1107       1107
hoenest                             929        929
sarcasm_only                        920        920
tartecosmetics                      873        873
ouatconfessions                     848        848
anastasiabeverlyhills               762        762
memequeen                           670        670
maccosmetics                        626        626
prettylittleliars                   579        579

This data tells me that my none of my top 10 liked accounts were about fashion, and my top liked account was the Cavs, as I have liked a post from the cavs account 1145 times as you can see from the "timestamp" column. I think there are some limitations to this process as I wish we could also filter out the information by year. This would give me the opportunity to get more concise data as my interest have switched throughout the years. Other data that I could use from my Instagram personal account files would be to group by the average account titles that I have liked posts from. 